src/java/org/apache/cassandra/tools/nodetool/Status.java

```diff
@@ -108,15 +156,32 @@ public void execute(NodeProbe probe)
             TableBuilder tableBuilder = sharedTable.next();
             addNodesHeader(hasEffectiveOwns, tableBuilder);
 
-            ArrayListMultimap<String, HostStatWithPort> hostToTokens = ArrayListMultimap.create();
+            ArrayListMultimap<InetAddressAndPort, HostStatWithPort> hostToTokens = ArrayListMultimap.create();
             for (HostStatWithPort stat : dc.getValue())
-                hostToTokens.put(stat.endpointWithPort.getHostAddressAndPort(), stat);
+                hostToTokens.put(stat.endpointWithPort, stat);
 
-            for (String endpoint : hostToTokens.keySet())
+            Map<String, List<Object>> data = new HashMap<>();
+            for (InetAddressAndPort endpoint : hostToTokens.keySet())
             {
-                Float owns = ownerships.get(endpoint);
+                Float owns = ownerships.get(endpoint.getHostAddressAndPort());
                 List<HostStatWithPort> tokens = hostToTokens.get(endpoint);
-                addNode(endpoint, owns, tokens.get(0), tokens.size(), hasEffectiveOwns, tableBuilder);
+
+                HostStatWithPort hostStatWithPort = tokens.get(0);
+                String epDns = hostStatWithPort.ipOrDns(printPort);
+                List<Object> nodeData = addNode(epDns, endpoint, owns, hostStatWithPort, tokens.size(), hasEffectiveOwns);
+                data.put(epDns, nodeData);
+            }
+
+            data = sort(data);
+
+            for (Map.Entry<String, List<Object>> entry : data.entrySet())
+            {
+                List<Object> values = entry.getValue();
+                List<String> row = new ArrayList<>();
+                for (int i = 1; i < values.size(); i++)
```

Comment:
We skip the first value (index 0) in the row because it is used solely for sorting, not for printing. The first value is an InetAddressAndPort, which cannot be cast to a String when printing, causing a ClassCastException. Instead, we print epDns (as returned by the addNode() function), which is a String and is appropriate for display in the table.